File: pkg/codegen/testing/utils/testdata/logical-name-pp/dotnet/logical-name.cs
using System.Collections.Generic;
using System.Linq;
using Pulumi;
using Random = Pulumi.Random;

return await Deployment.RunAsync(() => 
{
    var config = new Config();
    var configLexicalName = config.Require("cC-Charlie_charlie.😃⁉️");
    var resourceLexicalName = new Random.RandomPet("aA-Alpha_alpha.🤯⁉️", new()
    {
        Prefix = configLexicalName,
    });

    return new Dictionary<string, object?>
    {
        ["bB-Beta_beta.💜⁉"] = resourceLexicalName.Id,
        ["dD-Delta_delta.🔥⁉"] = resourceLexicalName.Id,
    };
});

File: tests/testdata/codegen/logical-name-pp/go/logical-name.go
package main

import (
	"github.com/pulumi/pulumi-random/sdk/v4/go/random"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		cfg := config.New(ctx, "")
		configLexicalName := cfg.Require("cC-Charlie_charlie.😃⁉️")
		resourceLexicalName, err := random.NewRandomPet(ctx, "aA-Alpha_alpha.🤯⁉️", &random.RandomPetArgs{
			Prefix: pulumi.String(pulumi.String(configLexicalName)),
		})
		if err != nil {
			return err
		}
		ctx.Export("bB-Beta_beta.💜⁉", resourceLexicalName.ID())
		ctx.Export("dD-Delta_delta.🔥⁉", resourceLexicalName.ID())
		return nil
	})
}
